python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -697,39 +889,164 @@ def setDefaults(self):
+    \anchor SafeClipAssembleCoaddTask
+
+    \brief Assemble a coadded image from a set of coadded temporary exposures, being careful to clip & flag areas 
+    with potential artifacts.
+
+    \section pipe_tasks_assembleCoadd_Contents Contents
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Purpose
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Initialize
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Run
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Config
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Debug
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Example
+
+    \section pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Purpose	Description
+
+    \copybrief SafeClipAssembleCoaddTask
+
+    We want to assemble a coadded image from a set of coadded temporary exposures (coaddTempExps).
+    We implicitly assume that the coaddTempExps were all observed in the same filter band.
+    Each input coaddTempExp covers a patch on the sky and corresponds to a single run/visit/exposure of the
+    covered patch.
+    We provide the task with a list of coaddTempExps (selectDataList) from which it selects coaddTempExps that
+    cover the specified patch (pointed at by dataRef).
+    Each coaddTempExp that goes into a coadd will typically have an independent photometric zero-point.
+    Therefore, we must scale each coaddTempExp to set it to a common photometric zeropoint.
+    By default, each coaddTempExp has backgrounds and hence will require config.doMatchBackgrounds=True.
+    When background matching is enabled, the task may be configured to automatically select a reference
+    exposure (config.autoReference=True).
+    If this is not done, we require that the input dataRef provides access to a coaddTempExp (dataset type
+    coaddName + 'Coadd_tempExp') which is used as the reference exposure.
+    The coadd is computed as a mean. This poses two problems: i. individual pixels rather than sources are
+    detected as outliers and discarded from the coadd - hence we can mistake genuine astronomical transients
+    for outliers, & ii. when computing the PSF at a given pixel on the coadd, the contribution from PSFs from
+    visits with outlier pixels (at that location on the coadd) rejected as outliers cannot be treated
+    correctly \ref AssebleCoaddTask_ "AssembleCoaddTask" does not keep track of outliers.
+    In this task, we correct for this behavior by creating a new badMaskPlane 'CLIPPED'. We populate this
+    plane on the input coaddTempExps and the final coadd where 1. difference imaging suggests that the
+    photometry is un-reliable and ii. there is insufficient overlap between the potential non-astronomical
+    artifact and a detected source. Such regions will not contribute to the final coadd. Furthermore, any
+    routine to determine the coadd PSF can now be cognizant of clipped regions. As in the case of the base
```

Comment:
I think there are a number of confusing things in this paragraph.  I think you need to clearly state the problem with AssembleCoadd using clipping and how SafeClip tries to fix this.